sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -641,12 +641,21 @@ bool ProgramManager::isSpecialDeviceImageShouldBeUsed(
     // TODO: re-design the encode of the devicelib metadata if we must support
     // more devicelib images in this way.
     enum { DEVICELIB_FALLBACK = 0, DEVICELIB_NATIVE };
+    ur_bool_t NativeBF16Supported = false;
     const std::shared_ptr<detail::device_impl> &DeviceImpl =
         detail::getSyclObjImpl(Dev);
-    std::string NativeBF16ExtName = "cl_intel_bfloat16_conversions";
-    bool NativeBF16Supported = (DeviceImpl->has_extension(NativeBF16ExtName));
-    return NativeBF16Supported ==
-           (Bfloat16DeviceLibVersion == DEVICELIB_NATIVE);
+    ur_result_t CallSuccessful =
```

Comment:
Hi, @steffenlarsen 
ur API returns ur_result_t value, if succeeds, the value is UR_RESULT_SUCCESS which is 0, assigning to a bool value will get "false".
Could you help review again? Thanks very much.

Reply:
Ah, good catch!